lua/snacks/git.lua

```diff
@@ -34,11 +38,20 @@ function M.get_root(path)
   -- check cache first
   for dir in vim.fs.parents(path) do
     if git_cache[dir] then
-      return vim.fs.normalize(dir) or nil
+      return git_cache[dir]
     end
   end
+
+  local git_root = get_git_root()
```

Comment:
why do you always execute `git` here?

Reply:
it'll be executed only if there are no parent folders with the cached git directory in the cache, which should probably happen only once as the lookup above will get the cached dir next time